### PR TITLE
mgr/dashboard: use bootstrap variables for colors

### PR DESCRIPTION
**Pull request by ishanrai05**
Fixes: https://tracker.ceph.com/issues/38891

Signed-off-by: Ishan Rai <ishanrai05@gmail.com>

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by LenzGr** [comment]
> Wouldn't be worthy to prevent this from ever happening again? What about adding [`sass-lint`](https://www.npmjs.com/package/sass-lint) with this rule?

I agree, this would be nice. Would you mind creating a tracker issue for this? I'm fine with either implementing this separately or by adding a separate change set that implements this in this PR.

**Comment by ishanrai05** [comment]
I am adding `sass-lint` to this PR

**Comment by epuertat** [comment]
Thanks, @ishanrai05!

BTW, it seems that while ago @pranav1698 opened a PR #34078 to tackle this very same issue. Anything in that PR that could be reused here?

**Comment by ishanrai05** [comment]
@epuertat  should we use variables for font sizes too?

**Comment by tspmelo** [comment]
@ishanrai05 if possible I would do that in another PR.
I have an issue (https://tracker.ceph.com/issues/38890) regarding the font size, where I suggest we refactor all our font sizes and use the new way they suggest in bootstrap 4.

**Comment by ishanrai05** [comment]
@tspmelo Yes. I am already working on that issue.

File: src/pybind/mgr/dashboard/frontend/src/styles/defaults.scss [hunk 1]
```diff
@@ -45,6 +46,8 @@ $color-light-gray: #d1d1d1 !default;
 $color-soft-gray: #ddd !default;
 $color-white-gray: #eee !default;
 $color-shade-gray: #efefef !default;
+$color-gray13: #212121;
+$color-gray20: #333333 !default;
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/src/styles/defaults.scss:50]: I just found out looking for a name in order to not name it myself  that https://www.color-hex.com/color/ shows gray13 and gray20 as the right color names for the colors. I first looked into another color naming system where both colors had the same name in the end, but I also asked myself would I have the slightest clue what color "Mine Shaft" or "Gray13" that would be? I bet not. Another thing to think about is that both colors are darker than our `dark-gray` color.

May be this is something to be discussed in another PR or an Issue. That said you don't have to find a solution for this in this PR. (At least from my side.)

Comment (ishanrai05) [reply src/pybind/mgr/dashboard/frontend/src/styles/defaults.scss:50]: Yes, I initially named it mine-shaft but then I changed it.

File: src/pybind/mgr/dashboard/frontend/package.json [hunk 2]
```diff
@@ -26,7 +26,8 @@
     "lint:prettier": "prettier --list-different \"{src,cypress}/**/*.{ts,scss}\"",
     "lint:html": "htmllint src/app/**/*.html && html-linter --config html-linter.config.json",
     "lint:tsc": "tsc -p src/tsconfig.app.json --noEmit && tsc -p tsconfig.spec.json --noEmit && tsc -p cypress/tsconfig.json --noEmit",
-    "lint": "npm run lint:tsc && npm run lint:tslint && npm run lint:prettier && npm run lint:html",
+    "lint:scss": "sass-lint -c ./.sasslintrc 'src/**/*.scss' -v -q",
+    "lint": "npm run lint:tsc && npm run lint:tslint && npm run lint:prettier && npm run lint:html && npm run lint:scss",
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/package.json:30]: As the number of lining checks is increasing, we could replace the sequential run with a parallel one ([npm-run-all](https://github.com/mysticatea/npm-run-all/blob/HEAD/docs/run-p.md)) and use some 'magic' to run all `lint:*` scripts:
```suggestion
    "lint": "run-p --continue-on-error --aggregate-output lint:*",
```

**Comment by ishanrai05** [comment]
@tspmelo Hi. Can you please suggest what other linting rules should I update?

**Comment by ishanrai05** [comment]
I will open another PR for scss linting.

**Comment by epuertat** [comment]
> I will open another PR for scss linting.

@ishanrai05 could you please indicate that new PR/tracker issue here? Thanks!

**Comment by ishanrai05** [comment]
@epuertat https://github.com/ceph/ceph/pull/35615

**Pull request merged**